### PR TITLE
Fixes for cross-compilation

### PR DESCRIPTION
**Pull request by Artturin**
the docs for bundlerApp used to have makeWrapper in buildInputs but it
has been corrected already

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/opa/default.nix [hunk 1]
```diff
@@ -44,8 +44,9 @@ stdenv.mkDerivation rec {
 
   configureFlags = [ "-ocamlfind ${ocamlPackages.findlib}/bin/ocamlfind" ];
 
+  nativeBuildInputs = [ makeWrapper ];
   buildInputs = [ which perl jdk openssl coreutils zlib ncurses
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/opa/default.nix:48]:
```suggestion
  buildInputs = [
    perl jdk openssl coreutils zlib ncurses
```

File: pkgs/development/compilers/opa/default.nix [hunk 2]
```diff
@@ -44,8 +44,9 @@ stdenv.mkDerivation rec {
...
-    makeWrapper gcc binutils gnumake nodejs
+    gcc binutils gnumake nodejs
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/opa/default.nix:49]:
```suggestion
```

File: pkgs/development/compilers/opa/default.nix [hunk 3]
```diff
@@ -44,8 +44,9 @@ stdenv.mkDerivation rec {
 
   configureFlags = [ "-ocamlfind ${ocamlPackages.findlib}/bin/ocamlfind" ];
 
+  nativeBuildInputs = [ makeWrapper ];
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/opa/default.nix:47]:
```suggestion
  nativeBuildInputs = [ gcc binutils nodejs makeWrapper which ];
```

File: pkgs/development/compilers/opendylan/default.nix [hunk 4]
```diff
@@ -13,8 +13,9 @@ stdenv.mkDerivation {
     fetchSubmodules = true;
   };
 
+  nativeBuildInputs = [ makeWrapper ];
   buildInputs = (if stdenv.hostPlatform.system == "i686-linux" then [ mps ] else [ boehmgc ]) ++ [
-    opendylan-bootstrap boehmgc gnused autoconf automake perl makeWrapper
+    opendylan-bootstrap boehmgc gnused autoconf automake perl
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/opendylan/default.nix:18]:
```suggestion
    opendylan-bootstrap boehmgc perl
```

File: pkgs/development/compilers/opendylan/default.nix [hunk 5]
```diff
@@ -13,8 +13,9 @@ stdenv.mkDerivation {
     fetchSubmodules = true;
   };
 
+  nativeBuildInputs = [ makeWrapper ];
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/opendylan/default.nix:16]:
```suggestion
  nativeBuildInputs = [ autoconf automake makeWrapper ];
```

File: pkgs/development/interpreters/lfe/generic-builder.nix [hunk 6]
```diff
@@ -37,7 +37,8 @@ buildRebar3 {
 
   inherit src version;
 
-  buildInputs = [ erlang makeWrapper ];
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ erlang ];
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/lfe/generic-builder.nix:41]: erlang is a compiler. It probably also goes to nativeBuildInputs.

File: pkgs/development/interpreters/maude/default.nix [hunk 7]
```diff
@@ -22,9 +22,9 @@ stdenv.mkDerivation {
     sha256 = "b112d7843f65217e3b5a9d40461698ef8dab7cbbe830af21216dfb924dc88a2f";
   };
 
-  nativeBuildInputs = [ unzip ];
+  nativeBuildInputs = [ unzip makeWrapper ];
   buildInputs = [
-    flex bison ncurses buddy tecla gmpxx libsigsegv makeWrapper cln yices
+    flex bison ncurses buddy tecla gmpxx libsigsegv cln yices
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/maude/default.nix:27]:
```suggestion
    ncurses buddy tecla gmpxx libsigsegv cln yices
```

File: pkgs/development/interpreters/maude/default.nix [hunk 8]
```diff
@@ -22,9 +22,9 @@ stdenv.mkDerivation {
     sha256 = "b112d7843f65217e3b5a9d40461698ef8dab7cbbe830af21216dfb924dc88a2f";
   };
 
-  nativeBuildInputs = [ unzip ];
+  nativeBuildInputs = [ unzip makeWrapper ];
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/maude/default.nix:25]:
```suggestion
  nativeBuildInputs = [ bison flex makeWrapper unzip ];
```

File: pkgs/development/mobile/androidenv/build-tools.nix [hunk 9]
```diff
@@ -2,8 +2,9 @@
 
 deployAndroidPackage {
   inherit package os;
-  buildInputs = [ makeWrapper ] ++
-    lib.optionals (os == "linux") [ autoPatchelfHook pkgs.glibc pkgs.zlib pkgs.ncurses5 pkgs_i686.glibc pkgs_i686.zlib pkgs_i686.ncurses5 pkgs.libcxx ];
+  nativeBuildInputs = [ makeWrapper ]
+    ++ lib.optionals (os == "linux") [ autoPatchelfHook ];
+  buildInputs = lib.optionals (os == "linux") [ pkgs.glibc pkgs.zlib pkgs.ncurses5 pkgs_i686.glibc pkgs_i686.zlib pkgs_i686.ncurses5 pkgs.libcxx ];
```
Comment (SuperSandro2000) [inline pkgs/development/mobile/androidenv/build-tools.nix:7]:
```suggestion
  buildInputs = lib.optionals (os == "linux") [ glibc libcxx ncurses5 zlib pkgs_i686.glibc pkgs_i686.zlib pkgs_i686.ncurses5 ];
```

File: pkgs/development/mobile/androidenv/tools/25.nix [hunk 10]
```diff
@@ -2,8 +2,8 @@
 
 deployAndroidPackage {
   name = "androidsdk";
-  buildInputs = [ autoPatchelfHook makeWrapper ]
-    ++ lib.optional (os == "linux") [ pkgs.glibc pkgs.xorg.libX11 pkgs.xorg.libXext pkgs.xorg.libXdamage pkgs.xorg.libxcb pkgs.xorg.libXfixes pkgs.xorg.libXrender pkgs.fontconfig.lib pkgs.freetype pkgs.libGL pkgs.zlib pkgs.ncurses5 pkgs.libpulseaudio pkgs_i686.glibc pkgs_i686.xorg.libX11 pkgs_i686.xorg.libXrender pkgs_i686.fontconfig pkgs_i686.freetype pkgs_i686.zlib ];
+  nativeBuildInputs = [ autoPatchelfHook makeWrapper ];
+  buildInputs = lib.optional (os == "linux") [ pkgs.glibc pkgs.xorg.libX11 pkgs.xorg.libXext pkgs.xorg.libXdamage pkgs.xorg.libxcb pkgs.xorg.libXfixes pkgs.xorg.libXrender pkgs.fontconfig.lib pkgs.freetype pkgs.libGL pkgs.zlib pkgs.ncurses5 pkgs.libpulseaudio pkgs_i686.glibc pkgs_i686.xorg.libX11 pkgs_i686.xorg.libXrender pkgs_i686.fontconfig pkgs_i686.freetype pkgs_i686.zlib ];
```
Comment (SuperSandro2000) [inline pkgs/development/mobile/androidenv/tools/25.nix:6]:
```suggestion
  buildInputs = lib.optional (os == "linux") [
    glibc xorg.libX11 xorg.libXext xorg.libXdamage xorg.libxcb xorg.libXfixes xorg.libXrender
    fontconfig.lib freetype libGL zlib ncurses5 libpulseaudio 
    pkgs_i686.glibc pkgs_i686.xorg.libX11 pkgs_i686.xorg.libXrender pkgs_i686.fontconfig 
    pkgs_i686.freetype pkgs_i686.zlib
  ];
```

File: pkgs/development/tools/build-managers/msbuild/default.nix [hunk 11]
```diff
@@ -36,12 +36,12 @@ stdenv.mkDerivation rec {
     dotnet-sdk
     mono
     unzip
+    makeWrapper
   ];
 
   buildInputs = [
     dotnetPackages.Nuget
     glibcLocales
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/msbuild/default.nix:44]:
```suggestion
    dotnet-sdk
    dotnetPackages.Nuget
    glibcLocales
    makeWrapper
    mono
    unzip
```

File: pkgs/games/megaglest/default.nix [hunk 12]
```diff
@@ -37,9 +37,9 @@ stdenv.mkDerivation {
     })
   ];
 
-  nativeBuildInputs = [ cmake pkg-config ];
+  nativeBuildInputs = [ cmake pkg-config makeWrapper ];
   buildInputs = [ git curl SDL2 xercesc openal lua libpng libjpeg libvlc wxGTK
```
Comment (SuperSandro2000) [inline pkgs/games/megaglest/default.nix:41]:
```suggestion
  buildInputs = [
    curl SDL2 xercesc openal lua libpng libjpeg libvlc wxGTK
```

File: pkgs/games/megaglest/default.nix [hunk 13]
```diff
@@ -37,9 +37,9 @@ stdenv.mkDerivation {
     })
   ];
 
-  nativeBuildInputs = [ cmake pkg-config ];
+  nativeBuildInputs = [ cmake pkg-config makeWrapper ];
```
Comment (SuperSandro2000) [inline pkgs/games/megaglest/default.nix:40]:
```suggestion
  nativeBuildInputs = [ cmake git pkg-config makeWrapper ];
```

File: pkgs/games/megaglest/default.nix [hunk 14]
```diff
@@ -37,9 +37,9 @@ stdenv.mkDerivation {
...
   buildInputs = [ git curl SDL2 xercesc openal lua libpng libjpeg libvlc wxGTK
-    glib cppunit fontconfig freetype ftgl glew libogg libvorbis makeWrapper libGLU ];
+    glib cppunit fontconfig freetype ftgl glew libogg libvorbis libGLU ];
```
Comment (SuperSandro2000) [inline pkgs/games/megaglest/default.nix:42]:
```suggestion
    glib cppunit fontconfig freetype ftgl glew libogg libvorbis libGLU
  ];
```

File: pkgs/games/xconq/default.nix [hunk 15]
```diff
@@ -10,8 +10,9 @@ stdenv.mkDerivation rec {
     sha256 = "1za78yx57mgwcmmi33wx3533yz1x093dnqis8q2qmqivxav51lca";
   };
 
+  nativeBuildInputs = [ makeWrapper ];
   buildInputs = [ cpio xorgproto libX11 libXmu libXaw libXt tcl tk libXext
```
Comment (SuperSandro2000) [inline pkgs/games/xconq/default.nix:14]:
```suggestion
  buildInputs = [
    cpio xorgproto libX11 libXmu libXaw libXt tcl tk libXext fontconfig
  ];
```

File: pkgs/games/xconq/default.nix [hunk 16]
```diff
@@ -10,8 +10,9 @@ stdenv.mkDerivation rec {
...
-    fontconfig makeWrapper ];
+    fontconfig ];
```
Comment (SuperSandro2000) [inline pkgs/games/xconq/default.nix:15]:
```suggestion
```

File: pkgs/tools/filesystems/glusterfs/default.nix [hunk 17]
```diff
@@ -16,7 +16,7 @@ let
   buildInputs = [
     fuse bison flex openssl ncurses readline
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/glusterfs/default.nix:17]:
```suggestion
    fuse openssl ncurses readline
```

File: pkgs/tools/filesystems/glusterfs/default.nix [hunk 18]
```diff
@@ -16,7 +16,7 @@ let
...
     libtool pkg-config zlib libaio libxml2
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/glusterfs/default.nix:18]:
```suggestion
    zlib libaio libxml2
```

File: pkgs/tools/filesystems/glusterfs/default.nix [hunk 19]
```diff
@@ -106,7 +106,7 @@ in stdenv.mkDerivation rec {
     "--localstatedir=/var"
   ];
 
-  nativeBuildInputs = [ rpcsvc-proto autoconf automake ];
+  nativeBuildInputs = [ rpcsvc-proto autoconf automake makeWrapper ];
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/glusterfs/default.nix:109]:
```suggestion
  nativeBuildInputs = [ autoconf automake bison flex libtool makeWrapper pkg-config rpcsvc-proto ];
```

File: pkgs/tools/security/wpscan/default.nix [hunk 20]
```diff
@@ -9,9 +9,7 @@ bundlerApp {
   gemdir = ./.;
   exes = [ "wpscan" ];
 
-  buildInputs = [
-    makeWrapper
-  ];
+  nativeBuildInputs = [ makeWrapper ];
```
Comment (SuperSandro2000) [inline pkgs/tools/security/wpscan/default.nix:12]: Please revert

File: pkgs/applications/emulators/cdemu/base.nix [hunk 21]
```diff
@@ -1,12 +1,12 @@
...
-{ lib, stdenv, fetchurl, cmake, pkg-config, buildInputs, drvParams ? {} }:
+attrs@{ lib, stdenv, fetchurl, cmake, pkg-config, buildInputs, nativeBuildInputs ? [], drvParams ? {} }:
 stdenv.mkDerivation ( rec {
   inherit pname version buildInputs;
   src = fetchurl {
     url = "mirror://sourceforge/cdemu/${pname}-${version}.tar.xz";
     sha256 = pkgSha256;
   };
-  nativeBuildInputs = [ pkg-config cmake ];
+  nativeBuildInputs = (attrs.nativeBuildInputs or [ ]) ++ [ pkg-config cmake ];
```
[the remark this answers is not in the document]
Comment (Artturin) [reply pkgs/applications/emulators/cdemu/base.nix:9]: nope
```
error: attribute 'nativeBuildInputs' missing

       at /nix/store/mzqhdx1sk6qxia1qvw7chvi8bbsjp9py-source/pkgs/applications/emulators/cdemu/base.nix:9:23:

            8|   };
            9|   nativeBuildInputs = attrs.nativeBuildInputs ++ [ pkg-config cmake ];
             |                       ^
           10|   setSourceRoot = ''
«derivation
```

if the `nativeBuildInputs ? []` is removed then 

```
nix-repl> cdemu-client
error: anonymous function at /nix/store/9czmzcsnb3lxk6dk1bv9j238s8yylzfl-source/pkgs/applications/emulators/cdemu/base.nix:2:1 called with unexpected argument 'nativeBuildInputs'

       at /nix/store/9czmzcsnb3lxk6dk1bv9j238s8yylzfl-source/lib/customisation.nix:69:16:

           68|     let
           69|       result = f origArgs;
             |                ^
           70|
```

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/emulators/cdemu/base.nix:9]: Not sure if that is nicer or not

File: pkgs/applications/gis/grass/default.nix [hunk 22]
```diff
@@ -15,9 +15,9 @@ stdenv.mkDerivation rec {
...
   };
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [ pkg-config makeWrapper ];
   buildInputs = [ flex bison zlib proj gdal libtiff libpng fftw sqlite
-  readline ffmpeg makeWrapper netcdf geos postgresql libmysqlclient blas
+  readline ffmpeg netcdf geos postgresql libmysqlclient blas
   libLAS proj-datumgrid zstd wrapGAppsHook ]
     ++ lib.optionals stdenv.isLinux [ cairo pdal wxGTK31 ]
     ++ lib.optional stdenv.isDarwin wxmac
```
[the remark this answers is not in the document]
Comment (Artturin) [reply pkgs/applications/gis/grass/default.nix:23]: the fails to apply lol

Leaving it for the maintainers @mpickering @willcohen

File: pkgs/tools/security/wpscan/default.nix (hunk 20 above)
Comment (Artturin) [reply pkgs/tools/security/wpscan/default.nix:12]: added support for nativeBuildInputs in [`3bfda6b` (#186564)](https://github.com/NixOS/nixpkgs/pull/186564/commits/3bfda6b18db43278f240d6598ba048c29375ec98)

File: pkgs/development/mobile/androidenv/tools/25.nix (hunk 10 above)
Comment (Artturin) [reply pkgs/development/mobile/androidenv/tools/25.nix:6]: Leaving it for the creator @svanderburg

File: pkgs/applications/science/math/pspp/default.nix [hunk 23]
```diff
@@ -13,10 +13,10 @@ stdenv.mkDerivation rec {
     sha256 = "0lqrash677b09zxdlxp89z6k02y4i23mbqg83956dwl69wc53dan";
   };
 
-  nativeBuildInputs = [ pkg-config texinfo python3 ];
+  nativeBuildInputs = [ pkg-config texinfo python3 makeWrapper ];
   buildInputs = [ libxml2 readline zlib perl cairo gtk3 gsl
```
[the remark this answers is not in the document]
Comment (Artturin) [reply pkgs/applications/science/math/pspp/default.nix:17]: no formatting pls, hopefully we will get a treewide format sometime..

File: pkgs/applications/emulators/cdemu/base.nix (hunk 21 above)
[the remark this answers is not in the document]
Comment (Artturin) [reply pkgs/applications/emulators/cdemu/base.nix:9]: @bendlas

**Review by ck3d — APPROVED** [review]
LGTM

**Pull request merged**